components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -75,6 +78,16 @@
      */
     public Application getApplication(String clientId) throws DCRMException {
 
+        try {
+            String tenantDomainOfApp = OAuth2Util.getTenantDomainOfOauthApp(clientId);
+            OAuth2Util.validateRequestTenantDomain(tenantDomainOfApp);
+        } catch (InvalidOAuthClientException e) {
+            throw new DCRMClientException(DCRMConstants.ErrorMessages.TENANT_DOMAIN_MISMATCH.getErrorCode(),
+                    String.format(DCRMConstants.ErrorMessages.TENANT_DOMAIN_MISMATCH.getMessage(), clientId));
+        } catch (IdentityOAuth2Exception e) {
+            throw new DCRMServerException(String.format(DCRMConstants.ErrorMessages.FAILED_TO_VALIDATE_TENANT_DOMAIN
+                    .getMessage(), clientId));
+        }
```

Comment:
Can we move this to a common private method and reuse?

Reply:
fixed